### PR TITLE
feat: add zk-solc-path option

### PR DESCRIPTION
**Pull request by elfedy**
## Motivation

[This PR ](https://github.com/Moonsong-Labs/compilers/pull/4) in zksync's fork of foundry compilers adds a new setting for zksolc's `--solc` option.

## Solution

Add `zk_solc_path` config value that allows the user to set the option

### CODE REVIEW

File: crates/cli/src/opts/build/mod.rs [hunk 1]
```diff
@@ -56,6 +58,15 @@ pub struct CompilerArgs {
     #[clap(help_heading = "Use ZKSync era vm", long)]
     pub zksync: bool,
 
+    #[clap(
+        help_heading = "zkSync Compiler options",
+        help = "Solc compiler path to when when compiling with zksolc",
```
Comment (Karrq) [inline crates/cli/src/opts/build/mod.rs:63]: nit: typo here (`when when`)

File: crates/cli/src/opts/build/mod.rs [hunk 2]
```diff
@@ -56,6 +58,15 @@ pub struct CompilerArgs {
...
+        long = "zk-solc-path",
+        value_name = "SYSTEM_MODE"
```
Comment (Karrq) [inline crates/cli/src/opts/build/mod.rs:65]: I think this might cause issues with the actual `system-mode` flag

**Review by Karrq — CHANGES_REQUESTED** [review]
I assume this is then used in `foundry-compilers`, right?

**Comment by elfedy** [comment]
> I assume this is then used in `foundry-compilers`, right?

@Karrq yeah its forwarded to foundry compilers when creating the ZkSolc config. Please see [associated PR](https://github.com/Moonsong-Labs/compilers/pull/4)

**Pull request merged**